idp/static/static.go

```diff
@@ -194,9 +197,11 @@ func (idp *identityProvider) Handle(ctx context.Context, w http.ResponseWriter,
 		id, err := idputil.HandleLoginForm(ctx, w, req, idpChoice, idp.initParams.Template, idp.loginUser)
 		if err != nil {
 			idp.initParams.VisitCompleter.RedirectFailure(ctx, w, req, ls.ReturnTo, ls.State, err)
+			return
 		}
 		if id != nil {
-			idp.initParams.VisitCompleter.RedirectSuccess(ctx, w, req, ls.ReturnTo, ls.State, id)
+			idp.initParams.VisitCompleter.RedirectMFA(ctx, w, req, idp.params.Require2FA, ls.ReturnTo, ls.State, state, id)
```

Comment:
This seems an odd way to do this, I assume RedirectMFA falls back to Redirect success if Require2FA is false?

Reply:
yes